lib/new_relic/agent.rb

```diff
@@ -649,6 +649,43 @@ def add_custom_span_attributes(params)
       end
     end
 
+    # Add custom attributes to log events for the current agent instance.
+    #
+    # @param [Hash] params    A Hash of attributes to attach to log
+    #                         events. The agent accepts up to 240 custom
+    #                         log event attributes.
+    #
+    #                         Keys will be coerced into Strings and must
+    #                         be less than 256 characters. Keys longer
+    #                         than 255 characters will be truncated.
+    #
+    #                         Values may be Strings, Symbols, numeric
+    #                         values or Booleans and must be less than
+    #                         4095 characters. If the value is a String
+    #                         or a Symbol, values longer than 4094
+    #                         characters will be truncated. If the value
+    #                         exceeds 4094 characters and is of a
+    #                         different class, the attribute pair will
```

Comment:
Yes, that's the intent of this phrase. 

Strings and symbols can be truncated. A value that isn't a string or symbol can't be truncated if it exceeds the limit. In that case, the key/value pair will not be sent. I think the "of a different class" was to provide a heads-up that floats, integers, etc. will just be dropped if they're too large.

Happy to change the phrasing to something that's clearer.

Reply:
I see now. The sentence as-is works great when paired with the sentence immediately preceding it.

Separate from value truncation.,. If the hash key itself is not on the allowlist of class types, does that key/value pair get dropped or perhaps the entire hash get rejected?

Reply:
At this time, all keys are coerced into strings before they're evaluated for truncation. This follows the logs API: all keys must be strings. If it can respond to `#to_s`, we'll make it a string for you.

Would you like me to stop coercing before truncation? 

I could see an issue if you had a key that was an integer that should be truncated. It'd be coerced into a string and then truncated with a different value that would be sent to New Relic.

Reply:
Got it. If length (not type) is the only concern because we're coercing and we're allowing 4095 for `#size`, I think we're ok.